### ISSUE TITLE
Object spread not transpiled correctly for Safari 11

### ISSUE THREAD

**Issue by filipesantoss** [babel/babel#10583]
## Bug Report

**Current Behavior**
Safari throwing error with message
```
Unexpected token '...'. Expected a property name.
```
 at 
```js
({ ...object })
```

**Input Code**

```js
const data = { name: 'username' };
const spread = object => ({ ...object, email: 'some@mail.com' });
const result = spread(data);

```

**Expected behavior/code**

`_objectSpread` to be called with `result` as a parameter, returning `{ name: 'username', email: 'some@mail.com' }`

**Babel Configuration (.babelrc, package.json, cli command)**

```js
{
    "presets": [
        [ "@babel/env", {  "useBuiltIns": "usage", "corejs": 3 } ],
        "@babel/react"
    ],
    "plugins": [
        "@babel/proposal-class-properties",
        [ "@babel/transform-runtime", { "corejs": 3 } ]
}
```

**Environment**
- Babel version(s): 7.6.4
- Node/npm version: 11.13.0
- OS: macOS High Sierra 10.13.2
- Monorepo: no
- How you are using Babel: loader

**Additional context/Screenshots**

Dependencies:

```
"@babel/plugin-proposal-class-properties": "^7.5.5",
"@babel/plugin-transform-runtime": "^7.6.2",
"@babel/preset-env": "^7.6.3",
"@babel/preset-react": "^7.6.3",
"@babel/runtime-corejs3": "^7.6.3",
"babel-loader": "^8.0.6",
"webpack": "^4.41.2",
"webpack-cli": "^3.3.9"
```
Webpack rule:

```
{
    test: /\.jsx?$/,
    exclude: /node_modules/,
    use: 'babel-loader'
}
```

Browserslist:

```
"> 0.25%, not dead, safari 11"
```

**Comment by JLHwung** [comment]
Could you pass `debug: true` to preset-env configurations? 
```json
[ "@babel/env", {  "useBuiltIns": "usage", "corejs": 3, "debug": true } ],
```
It will print the effective plugins if you re-run the webpack, which should look like
```
@babel/preset-env: `DEBUG` option

Using targets:
{
  "node": "12.12"
}

Using modules transform: false

Using plugins:
  syntax-async-generators { "node":"12.12" }
...
```
Could you attach it?

**Comment by filipesantoss** [comment]
Sure! Here it is:

```
Using targets:
{
  "chrome": "49",
  "edge": "17",
  "firefox": "68",
  "ie": "11",
  "ios": "11.3",
  "safari": "5.1",
  "samsung": "9.2"
}

Using modules transform: auto

Using plugins:
  transform-template-literals { "ie":"11", "ios":"11.3", "safari":"5.1" }
  transform-literals { "ie":"11", "safari":"5.1" }
  transform-function-name { "chrome":"49", "edge":"17", "ie":"11", "safari":"5.1" }
  transform-arrow-functions { "ie":"11", "safari":"5.1" }
  transform-block-scoped-functions { "safari":"5.1" }
  transform-classes { "ie":"11", "safari":"5.1" }
  transform-object-super { "ie":"11", "safari":"5.1" }
  transform-shorthand-properties { "ie":"11", "safari":"5.1" }
  transform-duplicate-keys { "ie":"11", "safari":"5.1" }
  transform-computed-properties { "ie":"11", "safari":"5.1" }
  transform-for-of { "chrome":"49", "ie":"11", "safari":"5.1" }
  transform-sticky-regex { "ie":"11", "safari":"5.1" }
  transform-dotall-regex { "chrome":"49", "edge":"17", "firefox":"68", "ie":"11", "safari":"5.1" }
  transform-unicode-regex { "chrome":"49", "ie":"11", "ios":"11.3", "safari":"5.1" }
  transform-spread { "ie":"11", "safari":"5.1" }
  transform-parameters { "edge":"17", "ie":"11", "safari":"5.1" }
  transform-destructuring { "chrome":"49", "ie":"11", "safari":"5.1" }
  transform-block-scoping { "ie":"11", "safari":"5.1" }
  transform-typeof-symbol { "ie":"11", "safari":"5.1" }
  transform-new-target { "ie":"11", "safari":"5.1" }
  transform-regenerator { "chrome":"49", "ie":"11", "safari":"5.1" }
  transform-exponentiation-operator { "chrome":"49", "ie":"11", "safari":"5.1" }
  transform-async-to-generator { "chrome":"49", "ie":"11", "safari":"5.1" }
  proposal-async-generator-functions { "chrome":"49", "edge":"17", "ie":"11", "ios":"11.3", "safari":"5.1" }
  proposal-object-rest-spread { "chrome":"49", "edge":"17", "ie":"11", "safari":"5.1" }
  proposal-unicode-property-regex { "chrome":"49", "edge":"17", "firefox":"68", "ie":"11", "safari":"5.1", "samsung":"9.2" }
  proposal-json-strings { "chrome":"49", "edge":"17", "ie":"11", "ios":"11.3", "safari":"5.1", "samsung":"9.2" }
  proposal-optional-catch-binding { "chrome":"49", "edge":"17", "ie":"11", "safari":"5.1", "samsung":"9.2" }
  transform-named-capturing-groups-regex { "chrome":"49", "edge":"17", "firefox":"68", "ie":"11", "safari":"5.1", "samsung":"9.2" }
  syntax-dynamic-import { "chrome":"49", "edge":"17", "firefox":"68", "ie":"11", "ios":"11.3", "safari":"5.1", "samsung":"9.2" }

```

**Comment by JLHwung** [comment]
I could not reproduce this error on [REPL](https://babeljs.io/repl#?babili=false&browsers=safari%2011&build=&builtIns=usage&spec=true&loose=false&code_lz=MYewdgzgLgBBAOAnApgQwCYwLwxAIwCtlhYsA-GACgG8YA6B_IkgGhmQFtUBLAGwC4YAcgggOyAAJc-dUByEwAvgEoA3EA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=false&presets=env&prettier=false&targets=&version=7.6.4&externalPlugins=), and from the log above we can see `proposal-object-rest-spread` is enabled. Could you provide a reproduction repo?

Another question: Could `Unexpected token '...'. Expected a property name.` be thrown from `node_modules`?

**Comment by filipesantoss** [comment]
After further inspection, I've found the source of the problem. Turns out we had another `ignore` regex on a webpack config that matched the name of the directory the troubling file was in.

There's no issue at all, this can be closed. Thanks for helping!

**Comment by JLHwung** [comment]
I am glad that you have solved your issue. It would be much easier to debug such configuration issues if we have implemented #10538
